File: dotnet/base/Mcma.Core/Logging/ConsoleLogger.cs
using System;
using Mcma.Core.Serialization;
using Newtonsoft.Json.Linq;

namespace Mcma.Core.Logging
{
    public class ConsoleLogger : Logger
    {
        public ConsoleLogger(string source, McmaTracker tracker = null)
            : base(source, tracker)
        {
        }

        protected override void Log(LogEvent logEvent)
        {
            if (logEvent.Level <= 0)
                return;

            if (logEvent.Level < 200)
                WriteToConsole(logEvent, ConsoleColor.Red);
            else if (logEvent.Level < 300)
                WriteToConsole(logEvent, ConsoleColor.Yellow);
            else
                WriteToConsole(logEvent);
        }

        private static void WriteToConsole(LogEvent logEvent, ConsoleColor? color = null)
        {
            var origColor = Console.ForegroundColor;
            try
            {
                if (color.HasValue)
                    Console.ForegroundColor = color.Value;

                var message =
                    string.Join("|",
                        logEvent.Timestamp.ToString("yyyy-MM-ddThh:mm:ss.fff"),
                        logEvent.Level,
                        logEvent.Source,
                        logEvent.Type,
                        logEvent.TrackerId ?? "(empty)",
                        logEvent.TrackerLabel ?? "(empty)",
                        logEvent.Message ?? "(empty)",
                        logEvent.Args != null ? JArray.FromObject(logEvent.Args).ToString() : "(empty)");

                Console.WriteLine(message);
            }
            finally
            {
                Console.ForegroundColor = origColor;
            }
        }
    }
}

File: nodejs/base/core/lib/logging/console-logger.ts
import { Logger } from "./logger";
import { McmaTrackerProperties } from "../model";
import { LogEvent } from "./log-event";

export class ConsoleLogger extends Logger {
    constructor(source: string, requestId?: string, tracker?: McmaTrackerProperties) {
        super(source, requestId, tracker);
    }

    log(logEvent: LogEvent): void {
        if (logEvent.level > 0) {
            if (logEvent.level <= 200) {
                console.error(logEvent.toString());
            } else if (logEvent.level < 400) {
                console.warn(logEvent.toString());
            } else {
                console.log(logEvent.toString());
            }
        }
    }
}

Logger.System = new ConsoleLogger("System");
